internal/compliance/evaluator.go

```
@@ -0,0 +1,277 @@
+}
+
+// recalculateOnly is the no-policies fallback that mirrors the PL/pgSQL
+// recalculate_device_compliance function: count compliance_results and
+// roll up to the device's denormalized status.
+func (e *Evaluator) recalculateOnly(ctx context.Context, q *store.Queries, deviceID string) error {
+	counts, err := q.ComplianceResultCounts(ctx, deviceID)
+	if err != nil {
+		return fmt.Errorf("compliance: count results for %s: %w", deviceID, err)
+	}
+
+	status := StatusUnknown
```

golangci-lint (2.12.2)

[error] ineffectual assignment to status

(ineffassign)